src/cascadia/TerminalSettingsModel/WslDistroGenerator.cpp

```diff
@@ -136,3 +143,183 @@ std::vector<Profile> WslDistroGenerator::GenerateProfiles()
+    {
+        if (distName.substr(0, std::min(distName.size(), DockerDistributionPrefix.size())) == DockerDistributionPrefix)
+        {
+            // Docker for Windows creates some utility distributions to handle Docker commands.
+            // Pursuant to GH#3556, because they are _not_ user-facing we want to hide them.
+            continue;
+        }
+
+        auto WSLDistro{ CreateDefaultProfile(distName) };
+
+        WSLDistro.Commandline(L"wsl.exe -d " + distName);
+        WSLDistro.DefaultAppearance().ColorSchemeName(L"Campbell");
+        WSLDistro.StartingDirectory(DEFAULT_STARTING_DIRECTORY);
+        WSLDistro.Icon(L"ms-appx:///ProfileIcons/{9acb9455-ca41-5af7-950f-6bca1bc9722f}.png");
+        profiles.emplace_back(WSLDistro);
+    }
+    return profiles;
+}
+
+// Function Description:
+// - Open the reg key the root of the WSL data, in HKCU\Software\Microsoft\Windows\CurrentVersion\Lxss
+// Arguments:
+// - <none>
+// Return Value:
+// - the HKEY if it exists and we can read it, else nullptr
+static HKEY openWslRegKey()
+{
+    HKEY hKey{ nullptr };
+    if (RegOpenKeyEx(HKEY_CURRENT_USER, RegKeyLxss, 0, KEY_ALL_ACCESS, &hKey) == ERROR_SUCCESS)
+    {
+        return hKey;
+    }
+    return nullptr;
+}
+
+// Function Description:
+// - Open the reg key for a single distro, underneath the root WSL key.
+// Arguments:
+// - wslRootKey: the HKEY for the Lxss node.
+// - guid: the string representation of the GUID for the distro to inspect
+// Return Value:
+// - the HKEY if it exists and we can read it, else nullptr
+static HKEY openDistroKey(const wil::unique_hkey& wslRootKey, const std::wstring& guid)
+{
+    HKEY hKey{ nullptr };
+    if (RegOpenKeyEx(wslRootKey.get(), guid.c_str(), 0, KEY_ALL_ACCESS, &hKey) == ERROR_SUCCESS)
+    {
+        return hKey;
+    }
+    return nullptr;
+}
+
+// Function Description:
+// - Get the list of all the guids of all the WSL distros from the registry. If
+//   we fail to open or read the root reg key, we'll return false.
+//   Places the guids of all the distros into the "guidStrings" param.
+// Arguments:
+// - wslRootKey: the HKEY for the Lxss node.
+// - names: a vector that receives all the guids of the installed distros.
+// Return Value:
+// - false if we failed to enumerate all the WSL distros
+static bool getWslGuids(const wil::unique_hkey& wslRootKey,
+                        std::vector<std::wstring>& guidStrings)
+{
+    if (!wslRootKey)
+    {
+        return false;
+    }
+
+    wchar_t buffer[39]; // a {GUID} is 38 chars long
+    for (DWORD i = 0;; i++)
+    {
+        DWORD length;
+        const auto result = RegEnumKeyEx(wslRootKey.get(), i, &buffer[0], &length, nullptr, nullptr, nullptr, nullptr);
+        if (result == ERROR_NO_MORE_ITEMS)
+        {
+            break;
+        }
+
+        if (result == ERROR_SUCCESS &&
+            length == 38 &&
+            buffer[0] == L'{' &&
+            buffer[37] == L'}')
+        {

```

Comment:
I'm chuckling at this loose guessing that it looks like and smells like a GUID without using an actual GUID parsing function like `StringFromGUID2` because you're just going to turn around and shove the string back at another registry API and don't actually need the GUID as a GUID.

Reply:
_lmao_

Reply:
Pretty funny how there's no `GUIDFromString`...
`IIDFromString` however could be used here if we want to. I just don't think it's necessary, because a string that's exactly 38 chars long and begins/ends with curly braces, looks and smells pretty significantly like a valid parameter. I'd probably rather remove the additional begin/end checks. 😄